### DISCUSSION TITLE
Dagger Shell: does it become the documented default interface? how far does it go?

### DISCUSSION THREAD

**Discussion by cwlbraa** [dagger/dagger#12135] [Ideas]
For context, I'm one of those devs, and I think there are many of us out there, that almost never uses interactive repls or shells outside of my highly-customized zsh and the occaisonal database query webgui frontend. 

I see the argument that folks are gonna write wrapper scripts for `call` regardless of SDK programmability, but is the idea here to build a featureful enough dagger shell that folks will be writing and committing dagger shell scripts as wrappers? a featureful enough shell that it works like a complete sdk with functions? how far do we intend to take the shell? and how can we get there incrementally?

@shykes @helderco @aluzzardi @samalba

**Comment by vikram-dagger** [comment]
Linking https://github.com/dagger/dagger/pull/9076#issuecomment-2508203939 as it has similar questions
